src/components/Map.jsx

```diff
@@ -1,34 +1,53 @@
+import "maplibre-gl/dist/maplibre-gl.css";
 import T from "prop-types";
 import MapGl, { Layer, Source } from "react-map-gl/maplibre";
-import "maplibre-gl/dist/maplibre-gl.css";
-import bbox from "@turf/bbox";
+import { useState, useEffect } from "react";
 import { AttributionControl } from "react-map-gl";
+import bbox from "@turf/bbox";
+import { mapLayersAtom } from "../atoms";
+import { useAtomValue } from "jotai";
 
-function Map({ artifact, mini }){
-  const isStatic = mini; // if mini is true, the map is static
-  let viewState = {
+/**
+ * Map component
+ * Children are the layers to render on the map
+ */
+function Map() {
+  const [viewState, setViewState] = useState({
     longitude: 0,
     latitude: 0,
     zoom: 0
-  };
+  });
+
+  const mapLayers = useAtomValue(mapLayersAtom);
+
+  // if there are layers, calculate the bounds
+  // each layer is a feature collection, so we need to calculate the bounds of all features
+  useEffect(() => {
+    if (mapLayers.length > 0) {
+      const bounds = mapLayers.reduce((acc, layer) => {
+        const layerBounds = bbox(layer);
+        return [
+          Math.min(acc[0], layerBounds[0]),
+          Math.min(acc[1], layerBounds[1]),
+          Math.max(acc[2], layerBounds[2]),
+          Math.max(acc[3], layerBounds[3])
+        ];
+      }, [Infinity, Infinity, -Infinity, -Infinity]);
 
-  if (artifact) {
-    const bounds = bbox(artifact);
-    viewState = {
-      longitude: (bounds[0] + bounds[2]) / 2,
-      latitude: (bounds[1] + bounds[3]) / 2,
-      zoom: 12
-    };
-  }
+      setViewState({
+        longitude: (bounds[0] + bounds[2]) / 2,
+        latitude: (bounds[1] + bounds[3]) / 2,
+        zoom: 12
+      });
```

Comment:
I'd suggest to use [fitBounds](https://maplibre.org/maplibre-gl-js/docs/API/classes/Map/#fitbounds), that way we can control the padding around the bounding box and we don't have to guess the right zoom level.